modules/indexer/issues/indexer.go

```diff
@@ -285,9 +254,6 @@ func InitIssueIndexer(syncReindex bool) {
 			case <-graceful.GetManager().IsShutdown():
 				log.Warn("Shutdown occurred before issue index initialisation was complete")
 			case <-time.After(timeout):
-				if shutdownable, ok := issueIndexerQueue.(queue.Shutdownable); ok {
```

Comment:
Why we don't need these lines any more?

Reply:
It doesn't make sense.

To resolve the "shutdown" problem,  Gitea should gracefully shutdown all queues by a general approach, but not only shutdown this one. The items in other queues would still be lost in the end.

Even if some items are lost, it's not that serous, next time editing the issue, it will be synced to queue again, that's still good enough.

Reply:
Yes, not only the queues shuting down but also all other services. Which is one aim of my services PR wants to resolve.

Reply:
It's impossible to have a clear solution at the moment: there are too many `Fatal` calls.

And there could always be some cases that items would be lost: for example: what if the redis server is down when Gitea is shutting down? The items on-the-fly couldn't be pushed back to the  queue any more.

Instead of making the system too complex, it's good to accept the fact that "there is no  transaction for queue, tolerate that some items could be lost in rare cases, the lost items could be re-queued from database".

Reply:
To support "shutdown", we can do this 3c98aa3f2f954073f99c01155efe49a7c0c9c1ca

(also added some comments about the "shutdown" problem)